File: app/src/main/java/com/alpacacore/example/chat/ChatMsg.java
package com.alpacacore.example.chat;

public class ChatMsg {
    public static enum Source {
        AI(0), USER(1);

        public final int value;

        private Source(int value) {
            this.value = value;
        }
    }
    Source src;
    String text;

    ChatMsg(Source src, String text) {
        this.src = src;
        this.text = text;
    }
}

File: app/src/main/java/com/alpacacore/example/chat/ChatMsg.kt
// Copyright (c) Alpaca Core
// SPDX-License-Identifier: MIT
//
package com.alpacacore.example.chat

data class ChatMsg(var src: Source, var text: String) {
    enum class Source(val value: Int) {
        AI(0), USER(1)
    }
}
